test/blackbox/api/dds-pim/TCPReqRepHelloWorldReplier.cpp

```diff
@@ -112,28 +136,53 @@ void TCPReqRepHelloWorldReplier::init(
         descriptor->tls_config.add_option(TLSOptions::NO_SSLV2);
     }
 
-    pattr.rtps.userTransports.push_back(descriptor);
+    participant_qos.transport().user_transports.push_back(descriptor);
 
-    participant_ = Domain::createParticipant(pattr);
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(
+        domainId, participant_qos);
     ASSERT_NE(participant_, nullptr);
+    ASSERT_TRUE(participant_->is_enabled());
 
     // Register type
-    ASSERT_EQ(Domain::registerType(participant_, &type_), true);
+    type_.reset(new HelloWorldPubSubType());
+    ASSERT_EQ(participant_->register_type(type_), ReturnCode_t::RETCODE_OK);
+
+    configDatareader("Request");
+    request_topic_ = participant_->create_topic(datareader_topicname_,
+                    type_->getName(), TOPIC_QOS_DEFAULT);
+    ASSERT_NE(request_topic_, nullptr);
+    ASSERT_TRUE(request_topic_->is_enabled());
+
+    configDatawriter("Reply");
+    reply_topic_ = participant_->create_topic(datawriter_topicname_,
+                    type_->getName(), TOPIC_QOS_DEFAULT);
+    ASSERT_NE(reply_topic_, nullptr);
+    ASSERT_TRUE(reply_topic_->is_enabled());
 
     //Create subscriber
-    sattr.topic.topicKind = NO_KEY;
-    sattr.topic.topicDataType = type_.getName();
-    configSubscriber("Request");
-    request_subscriber_ = Domain::createSubscriber(participant_, sattr, &request_listener_);
+    request_subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT);
     ASSERT_NE(request_subscriber_, nullptr);
+    ASSERT_TRUE(request_subscriber_->is_enabled());
 
     //Create publisher
-    puattr.topic.topicKind = NO_KEY;
-    puattr.topic.topicDataType = type_.getName();
-    puattr.topic.topicName = "HelloWorldTopicReply";
-    configPublisher("Reply");
-    reply_publisher_ = Domain::createPublisher(participant_, puattr, &reply_listener_);
+    reply_publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT);
     ASSERT_NE(reply_publisher_, nullptr);
+    ASSERT_TRUE(reply_publisher_->is_enabled());
+
+    //Create datareader
+    datareader_qos_.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    datareader_qos_.reliability().max_blocking_time = Duration_t(1, 0);
+    request_datareader_ = request_subscriber_->create_datareader(request_topic_, datareader_qos_,
+                    &request_listener_);
+    ASSERT_NE(request_datareader_, nullptr);
+    ASSERT_TRUE(request_datareader_->is_enabled());
+
+    //Create datawriter
+    datawriter_qos_.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    datawriter_qos_.reliability().max_blocking_time = Duration_t(1, 0);
```

Comment:
I think we should keep the comment just to ensure its maintainability

```suggestion
    datawriter_qos_.reliability().kind = RELIABLE_RELIABILITY_QOS;
    // Increase default max_blocking_time to 1 second, as our CI infrastructure shows some
    // big CPU overhead sometimes
    datawriter_qos_.reliability().max_blocking_time = Duration_t(1, 0);
```